x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/connector/api_key_configuration.tsx

```diff
@@ -122,6 +122,43 @@ export const ApiKeyConfig: React.FC<{
               )}
         </EuiText>
       </EuiFlexItem>
+      {!isNative || status === Status.LOADING ? (
+        <></>
+      ) : indexName === '' ? (
+        <EuiCallOut
+          iconType="iInCircle"
+          title={i18n.translate(
+            'xpack.enterpriseSearch.content.connector_detail.configurationConnector.nativeConnector.apiKey.waitingForAttachedIndex',
+            {
+              defaultMessage:
+                'An API key will be automatically generated when an index is attached to this connector.',
+            }
+          )}
+        />
+      ) : hasApiKey ? (
+        <EuiCallOut
+          iconType="check"
+          color="success"
+          title={i18n.translate(
+            'xpack.enterpriseSearch.content.connector_detail.configurationConnector.nativeConnector.apiKey.exists',
+            {
+              defaultMessage: `An API has been generated for this connector.`,
+            }
+          )}
+        />
+      ) : (
+        <EuiCallOut
+          iconType="warning"
+          color="danger"
+          title={i18n.translate(
+            'xpack.enterpriseSearch.content.connector_detail.configurationConnector.nativeConnector.apiKey.missing',
+            {
+              defaultMessage:
+                'This connector is missing an API key. Click "Generate API key" to regenerate it.',
```

Comment:
nitpick: we generally don't want to say "click X" because people may be using keyboard or other means that don't actually click buttons. @leemthompo should have better copy suggestions.

Reply:
I removed the whole second sentence. The user can understand from the UI that there's a big "Generate API key" button to use.